### PR TITLE
feature/addPriorityControllerInfrastructure

### PR DESCRIPTION
**Pull request by Glunky**

### CODE REVIEW

File: src/DashboardService.Business/Priority/CreatePriorityCommand.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+using LT.DigitalOffice.DashboardService.Models.Dto.Requests.Priority;
+using LT.DigitalOffice.Kernel.Responses;
+using System;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Business.Priority;
+
+public class CreatePriorityCommand : ICreatePriorityCommand
+{
+  private readonly PriorityRepository _priorityRepository;
```
Comment (Glunky) [inline src/DashboardService.Business/Priority/CreatePriorityCommand.cs]: replace with interface for all commands

File: src/DashboardService.Mappers/Models/Interfaces/IPriorityInfoMapper.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+using LT.DigitalOffice.DashboardService.Models.Db;
+using LT.DigitalOffice.DashboardService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.DashboardService.Mappers.Models.Interfaces;
+
+public interface IPriorityInfoMapper
```
Comment (Glunky) [inline src/DashboardService.Mappers/Models/Interfaces/IPriorityInfoMapper.cs:8]: add autoinject

File: src/DashboardService.Business/Priority/CreatePriorityCommand.cs (hunk 1 above)
Comment (SolovyovDmitry) [reply src/DashboardService.Business/Priority/CreatePriorityCommand.cs]: self review?

File: src/DashboardService.Business/Priority/Interfaces/IGetAllPrioritiesCommand.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+using LT.DigitalOffice.DashboardService.Models.Dto.Models;
+using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.Kernel.Responses;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Business.Priority.Interfaces;
+
+[AutoInject]
+public interface IGetAllPrioritiesCommand
```
Comment (SolovyovDmitry) [inline src/DashboardService.Business/Priority/Interfaces/IGetAllPrioritiesCommand.cs]: just GetPriorities

File: src/DashboardService.Business/Priority/Interfaces/IGetAllPrioritiesCommand.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+{
+  Task<FindResultResponse<IEnumerable<PriorityInfo>>> ExecuteAsync();
```
Comment (SolovyovDmitry) [inline src/DashboardService.Business/Priority/Interfaces/IGetAllPrioritiesCommand.cs]: Make T = PriorityInfo. Watch implementation

File: src/DashboardService.Business/Priority/GetAllPrioritiesCommand.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+public class GetAllPrioritiesCommand : IGetAllPrioritiesCommand
+{
+  private readonly PriorityRepository _priorityRepository;
+
+  public GetAllPrioritiesCommand(PriorityRepository priorityRepository)
+  {
+    _priorityRepository = priorityRepository;
+  }
+  
+  public Task<FindResultResponse<IEnumerable<PriorityInfo>>> ExecuteAsync()
```
Comment (SolovyovDmitry) [inline src/DashboardService.Business/Priority/GetAllPrioritiesCommand.cs]: provide filtering model which inherits BaseFindFilter

File: src/DashboardService.Models.Dto/Responses/PriorityResponse.cs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+using System;
+
+namespace LT.DigitalOffice.DashboardService.Models.Dto.Responses;
+
+public class PriorityResponse
```
Comment (SolovyovDmitry) [inline src/DashboardService.Models.Dto/Responses/PriorityResponse.cs]: It's so simple that I guess PriorityInfo is enough

File: src/DashboardService.Data/Interfaces/IPriorityRepository.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Data.Interfaces;
+
+[AutoInject]
+public interface IPriorityRepository
+{
+  Task<Guid?> CreateAsync(DbPriority dbDepartment);
+  Task<IEnumerable<DbPriority>> GetAllAsync();
```
Comment (SolovyovDmitry) [inline src/DashboardService.Data/Interfaces/IPriorityRepository.cs]: Just List. I don't think you will handle all the generics

File: src/DashboardService.Data/PriorityRepository.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Data;
+
+public class PriorityRepository : IPriorityRepository
+{
+  private readonly IDataProvider _provider;
+
+  public PriorityRepository([FromServices] IDataProvider provider)
```
Comment (SolovyovDmitry) [inline src/DashboardService.Data/PriorityRepository.cs]: You inject it with constructor so it's useless here

File: src/DashboardService.Models.Dto/Requests/Priority/CreatePriorityRequest.cs [hunk 9]
```diff
@@ -0,0 +1,6 @@
+namespace LT.DigitalOffice.DashboardService.Models.Dto.Requests.Priority;
+
+public record CreatePriorityRequest
+{
+  public string PriorityName { get; set; }
```
Comment (SolovyovDmitry) [inline src/DashboardService.Models.Dto/Requests/Priority/CreatePriorityRequest.cs]: Just Name

File: src/DashboardService.Validation/Priority/CreatePriorityRequestValidator.cs [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+using LT.DigitalOffice.DashboardService.Models.Dto.Requests.Priority;
+using LT.DigitalOffice.DashboardService.Validation.Priority.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace LT.DigitalOffice.DashboardService.Validation.Priority;
+
+public class CreatePriorityRequestValidator : AbstractValidator<CreatePriorityRequest>, ICreatePriorityRequestValidator
+{
+
+  public CreatePriorityRequestValidator([FromServices] IPriorityRepository reposisoty)
```
Comment (SolovyovDmitry) [inline src/DashboardService.Validation/Priority/CreatePriorityRequestValidator.cs]: use this attribute only in Controller

File: src/DashboardService.Validation/Priority/Interfaces/ICreatePriorityRequestValidator.cs [hunk 11]
```diff
@@ -0,0 +1,8 @@
+using FluentValidation;
+using LT.DigitalOffice.DashboardService.Models.Dto.Requests.Priority;
+using LT.DigitalOffice.Kernel.Attributes;
+
+namespace LT.DigitalOffice.DashboardService.Validation.Priority.Interfaces;
+
+[AutoInject]
+public interface ICreatePriorityRequestValidator : IValidator<CreatePriorityRequest> { }
```
Comment (SolovyovDmitry) [inline src/DashboardService.Validation/Priority/Interfaces/ICreatePriorityRequestValidator.cs:8]: can {} be replaced with ';' ?

File: src/DashboardService/Controllers/PriorityController.cs [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Controllers;
+
+[Route("[controller]")]
+[ApiController]
+public class PriorityController : ControllerBase
```
Comment (SolovyovDmitry) [inline src/DashboardService/Controllers/PriorityController.cs]: PrioritiesController

File: src/DashboardService/Controllers/PriorityController.cs [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+{
+  [HttpPost("create")]
```
Comment (SolovyovDmitry) [inline src/DashboardService/Controllers/PriorityController.cs]: drop this from routes. Make it like in HWs

File: src/DashboardService/Controllers/PriorityController.cs [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+    [FromRoute] Guid id,
+    [FromServices] GetPriorityCommand command)
+  {
+    return await command.ExecuteAsync(id);
+  }
+
+  [HttpPatch("edit/{id}")]
+  public async Task<OperationResultResponse<bool>> Patch(
+    [FromRoute] Guid id, 
+    [FromBody] PatchPriorityRequest request,
```
Comment (SolovyovDmitry) [inline src/DashboardService/Controllers/PriorityController.cs]: JsonPatchDocumentT

**Review by SolovyovDmitry — CHANGES_REQUESTED** [review]

File: src/DashboardService.Business/Priority/CreatePriorityCommand.cs (hunk 1 above)
Comment (Glunky) [reply src/DashboardService.Business/Priority/CreatePriorityCommand.cs]: )) ++

File: src/DashboardService.Mappers/Models/Interfaces/IPriorityInfoMapper.cs (hunk 2 above)
Comment (Glunky) [reply src/DashboardService.Mappers/Models/Interfaces/IPriorityInfoMapper.cs:8]: ++

File: src/DashboardService.Business/Priority/Interfaces/IGetAllPrioritiesCommand.cs (hunk 3 above)
Comment (Glunky) [reply src/DashboardService.Business/Priority/Interfaces/IGetAllPrioritiesCommand.cs]: ++

File: src/DashboardService.Business/Priority/Interfaces/IGetAllPrioritiesCommand.cs (hunk 4 above)
Comment (Glunky) [reply src/DashboardService.Business/Priority/Interfaces/IGetAllPrioritiesCommand.cs]: ++

File: src/DashboardService.Models.Dto/Responses/PriorityResponse.cs (hunk 6 above)
Comment (Glunky) [reply src/DashboardService.Models.Dto/Responses/PriorityResponse.cs]: ++

File: src/DashboardService.Data/Interfaces/IPriorityRepository.cs (hunk 7 above)
Comment (Glunky) [reply src/DashboardService.Data/Interfaces/IPriorityRepository.cs]: ++

File: src/DashboardService.Models.Dto/Requests/Priority/CreatePriorityRequest.cs (hunk 9 above)
Comment (Glunky) [reply src/DashboardService.Models.Dto/Requests/Priority/CreatePriorityRequest.cs]: ++

File: src/DashboardService.Validation/Priority/CreatePriorityRequestValidator.cs (hunk 10 above)
Comment (Glunky) [reply src/DashboardService.Validation/Priority/CreatePriorityRequestValidator.cs]: ++

File: src/DashboardService.Validation/Priority/Interfaces/ICreatePriorityRequestValidator.cs (hunk 11 above)
Comment (Glunky) [reply src/DashboardService.Validation/Priority/Interfaces/ICreatePriorityRequestValidator.cs:8]: no

File: src/DashboardService/Controllers/PriorityController.cs (hunk 12 above)
Comment (Glunky) [reply src/DashboardService/Controllers/PriorityController.cs]: ++

File: src/DashboardService/Controllers/PriorityController.cs (hunk 13 above)
Comment (Glunky) [reply src/DashboardService/Controllers/PriorityController.cs]: ++

File: src/DashboardService/Controllers/PriorityController.cs (hunk 14 above)
Comment (Glunky) [reply src/DashboardService/Controllers/PriorityController.cs]: ++

**Pull request closed without merging**